src/main/java/jp/co/soramitsu/sora/common/ArrayTree.java

```diff
@@ -0,0 +1,96 @@
+package jp.co.soramitsu.sora.common;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Getter;
+
+public class ArrayTree<E> {
```

Comment:
why do we need to implement it by yourself? I assume, there are some implementations of it. Am I right?

Reply:
I did not find any good implementation. If you point me to any good implementation, I will replace this.